lib/rules/no-test-module-for.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const types = require('../utils/types');
+const emberUtils = require('../utils/ember');
+const utils = require('../utils/utils');
+
+const ERROR_MESSAGE = 'moduleFor* apis are are not allowed. Use `module` instead of `moduleFor`';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Disallow use of moduleFor, moduleForComponent, etc',
+      category: 'Testing',
+      recommended: false,
+    },
+    fixable: null,
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    const filename = context.getFilename();
+    const isTestFile =
+      emberUtils.isTestFile(filename) ||
+      (filename.includes('tests') && filename.includes('helpers'));
+
+    if (!isTestFile) {
+      return {};
+    }
+
+    return {
+      FunctionDeclaration(node) {
+        if (isModuleFor(context, node)) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+      },
+      CallExpression(node) {
+        if (isModuleFor(context, node)) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+      },
+    };
+  },
+};
+
+function isModuleFor(context, node) {
+  const { callee, id } = node; // CallExpression | FunctionDeclaration
+  const identifier = callee || id;
+
+  if (!identifier || !types.isIdentifier(identifier)) {
+    return false;
+  }
+
+  // catches everything from legacy ember-qunit
+  const isLegacyApi =
+    identifier.name.startsWith('moduleFor') &&
+    utils.getSourceModuleNameForIdentifier(context, identifier) === 'ember-qunit';
```

Comment:
This import check is bypassed by the `isBespokeInvocation` check a few lines later, so it doesn't seem worthwhile to keep it anymore. Shall we just remove it at this point? (it's also a slightly expensive check, and we would need to add import statement tests if we were to keep it)